packages/compass-collection/src/modules/collection-tab.ts

```diff
@@ -35,7 +35,13 @@ import { calculateSchemaDepth } from '../calculate-schema-depth';
 import { processSchema } from '../transform-schema-to-field-info';
 import type { Document, MongoError } from 'mongodb';
 import { MockDataGeneratorStep } from '../components/mock-data-generator-modal/types';
-import type { MockDataGeneratorState } from '../components/mock-data-generator-modal/types';
+import type {
+  FakerSchemaMapping,
+  MockDataGeneratorState,
+} from '../components/mock-data-generator-modal/types';
+
+/* eslint-disable @typescript-eslint/no-require-imports */
+const faker = require('@faker-js/faker/locale/en');
```

Comment:
The root cause of the error you are seeing is that tsc for this package is targeting commonjs, on practice this doesn't really matter, so you can continue using import instead of switching it to require and add a `expect-error` notation instead of disabling an eslint rule

```suggestion
// @ts-expect-error TypeScript warns us about importing esm when output is cjs,
// but in reality this code will be just consumed by webpack, so it doesn't
// really matter
import { faker } from '@faker-js/faker/locale/en';
```

The benefits of that are that you still have proper types for the package (using require just completely bails out of the module resolution and so you just get `any` instead) and when compass repo fully supports esm and we are not targeting commonjs anymore, compiler will highlight this code (as it won't be an error any more) and we can clean it up.

Also generally speaking, I would strongly suggest to document why rules are being broken when they are broken 🙂 It is somewhat unavoidable sometimes to do so, indeed, but leaving the context is a good practice for a maintainable codebase